spec/consumer_set_spec.rb

```diff
@@ -60,19 +69,26 @@ def subscription(name)
 
         it "returns remaining messages of current partition" do
           config.fetch_messages = 1000
-          @messages = [:msg1, :msg2, Rdkafka::RdkafkaError.new(-191), :msgN]
-          allow(rdconsumer).to receive(:poll) do
-            message = @messages.shift
-            message.is_a?(StandardError) ? raise(message) : message
-          end
+          messages = [:msg1, :msg2, partition_eof_error, :msgN]
+          allow(rdconsumer).to receive(:poll, &message_generator(messages))
 
           expect(consumer_set.batch_poll(100)).to eq [:msg1, :msg2]
         end
 
-        it "drops nil messages" do
-          config.fetch_messages = 7
-          allow(rdconsumer).to receive(:poll).and_return(nil, nil, :msg1, nil, :msg2, nil, :msg3)
-          expect(consumer_set.batch_poll(100)).to eq [:msg1, :msg2, :msg3]
+        it "returns messages until nil is encountered" do
+          config.fetch_messages = 3
+          allow(rdconsumer).to receive(:poll).and_return(:msg1, :msg2, nil, :msg3)
+          expect(consumer_set.batch_poll(100)).to eq [:msg1, :msg2]
+        end
+
+        it "eventually reads all messages" do
+          config.fetch_messages = 1
+          messages = [:msg1, :msg2, nil, nil, partition_eof_error, partition_eof_error,  :msgN]
+          allow(rdconsumer).to receive(:poll, &message_generator(messages))
+
+          polled = []
+          messages.size.times { polled += consumer_set.batch_poll(100) }
+          expect(polled).to eq [:msg1, :msg2, :msgN]
```

Comment:
This is a change in behavior. A single `batch_poll` returns messages from multiple partitions. To be more consistent with the previous behavior of Racecar this behavior should be kept.